xbmc/pvr/timers/PVRTimerInfoTag.cpp

```diff
@@ -976,36 +976,35 @@ std::string CPVRTimerInfoTag::GetDeletedNotificationText() const
   return StringUtils::Format("%s: '%s'", g_localizeStrings.Get(stringID).c_str(), m_strTitle.c_str());
 }
 
-CEpgInfoTagPtr CPVRTimerInfoTag::GetEpgInfoTag(void) const
+CEpgInfoTagPtr CPVRTimerInfoTag::GetEpgInfoTag(bool bCreate /* = true */) const
 {
-  CPVRChannelPtr channel;
-  if (!m_epgTag)
-    channel = g_PVRChannelGroups->GetByUniqueID(m_iClientChannelUid, m_iClientId);
-
-  if (channel)
+  if (!m_epgTag && bCreate)
   {
-    CSingleLock lock(m_critSection);
-    if (!m_epgTag)
+    CPVRChannelPtr channel(g_PVRChannelGroups->GetByUniqueID(m_iClientChannelUid, m_iClientId));
+    if (channel)
     {
       const CEpgPtr epg(channel->GetEPG());
       if (epg)
       {
-        if (m_iEpgUid != EPG_TAG_INVALID_UID)
-        {
-          m_epgTag = epg->GetTagByBroadcastId(m_iEpgUid);
-        }
-        else if (!m_bStartAnyTime && !m_bEndAnyTime)
+        CSingleLock lock(m_critSection);
+        if (!m_epgTag)
```

Comment:
Hehe we had that already. This is the double checked locking pattern. First check needs to be done without muted locked. Between first and second check (this time with the lock) the member value might have changed.